### PR TITLE
Remove unused and optimize imports

### PR DESCRIPTION
**Pull request by philvarner**
There are a large number of unused import warnings, this PR resolves all of them. This was done using IntelliJ's Optimize Imports, which also reorders imports. No specific imports were collapsed into `_`.

Draft PR for now, as this contains the build fixes in https://github.com/sksamuel/elastic4s/pull/2024

### CODE REVIEW

**Comment by sksamuel** [comment]
2024 is merged. Let me know when you want this merging in.

**Comment by philvarner** [comment]
Ready for merge.

File: .sbtopts [hunk 1]
```diff
@@ -1,4 +1,3 @@
 -J-Xmx4G
 -J-XX:MaxMetaspaceSize=1G
--J-XX:MaxPermSize=1G
```
Comment (philvarner) [inline .sbtopts]: this flag is deprecated in OpenJDK 8

File: build.sbt [hunk 2]
```diff
@@ -30,10 +30,11 @@ val ZIOVersion             = "1.0.0-RC18"
 val SprayJsonVersion       = "1.3.5"
 val SttpVersion            = "1.7.2"
 val Slf4jVersion           = "1.7.30"
+val ScalatestPlusMockitoArtifactId = "mockito-3-2"
 
 lazy val commonScalaVersionSettings = Seq(
-  scalaVersion := "2.12.8",
```
Comment (philvarner) [inline build.sbt]: update to latest scala versions

**Pull request merged**